In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data = pd.read_csv("https://raw.githubusercontent.com/marsgr6/EN-online/8a1cee296279c274d8f05f3905fce97c1f7636a2/data/UX_data.csv")

In [5]:
Solucion pregunta 1

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   SUS     30 non-null     float64
 1   Time    30 non-null     float64
 2   ID      30 non-null     object 
 3   Gender  30 non-null     object 
 4   Age     30 non-null     object 
 5   Kinect  30 non-null     object 
 6   NUI     30 non-null     object 
 7   Games   30 non-null     object 
dtypes: float64(2), object(6)
memory usage: 2.0+ KB


Solucion pregunta 1.
data.info()  Usando la funcion info() podemos obsever todas las variables numericas en este caso: SUS y Time son float

In [8]:
data.info()

data[['SUS','Time']].describe() #escogemos aquellos valores que son numericos y usando la funcion describe obtenemos el resultado.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   SUS     30 non-null     float64
 1   Time    30 non-null     float64
 2   ID      30 non-null     object 
 3   Gender  30 non-null     object 
 4   Age     30 non-null     object 
 5   Kinect  30 non-null     object 
 6   NUI     30 non-null     object 
 7   Games   30 non-null     object 
dtypes: float64(2), object(6)
memory usage: 2.0+ KB


SUS        Time
count  30.000000   30.000000
mean   76.083333  157.500000
std    14.616997   22.399415
min    40.000000  112.700000
25%    68.125000  140.025000
50%    78.750000  159.700000
75%    87.500000  177.050000
max    95.000000  188.300000

Interpretacion de los cuartiles.
SUS:
Los cuartiles obtenidos dividen a la muestra en cuatro partes, el primer cuartil significa que 25% de los datos es menor o igual a 68.18. En nuestro estudio significa que el 25% de los individuos calificaron la usabilidad de ePHoRt con un valor menor o igual a 68.18.

El tercer cuartil nos informa que un 25% de los individuos dieron una calificacion mayor o igual a 87.5.

Ademas poemos apreciar que en la usabilidad del sistema la mediana es de 78.5 y esta muy cerca de la media. Basandome en la devivacion standar(14.61), los datos tienen un distribucion normal.


Time:(tiempo de ralizacion de las tareas)

En el tiempo utilizado para realizar las tareas podemos aprecias que a la mitad de la poblacion le tomo entre 78.75 y 95 en completar el trabajo. Ademas el 25% de los individuos duraron menos de 68.125, siendo el mejor tiempo de 40 y el maximo de 95.




2. (1 pt) Variables categóricas:

In [11]:
data['ID'].value_counts()
data['Gender'].value_counts()
data['Age'].value_counts()

twenty    17
thirty     7
forty      6
Name: Age, dtype: int64

3. (1 pt) Descripción de variables numéricas agrupadas:

In [18]:
data.groupby('Gender')['SUS','Time'].describe()

/tmp/ipykernel_8357/1356082860.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  data.groupby('Gender')['SUS','Time'].describe()


SUS                                                     Time  \
       count    mean        std   min     25%   50%   75%   max count   
Gender                                                                  
F       10.0  74.000  17.328525  40.0  66.250  77.5  87.5  92.5  10.0   
M       20.0  77.125  13.432948  42.5  69.375  80.0  87.5  95.0  20.0   

                                                                   
           mean        std    min      25%     50%     75%    max  
Gender                                                             
F       172.190  15.656341  144.0  165.975  175.50  184.75  188.3  
M       150.155  21.893582  112.7  135.600  148.65  167.30  185.3

Medias de Centralidad
Media y la media(mean)
Como primer resultado podemos apreciar que el 33 de los individuos pertenece al genero femenino y el 66% al genero masculino.
Ambos generos comparten un resultado similar en la media y mediana respecto a la usabilidad del sistema es decir no existe una diferencia significativa respecto a la calificacion de la usabilidad del sistema, esto se ve reflefajado en la desvicion standar al igual que sus cuartiles. A diferencia de los resultados obtenidos en el tiempo requerido para completar la tarea podemos apreciar que la media y la median del genero masculino difiere con un valor considerablemente menor al del genero femenino. Las medidas de dispercion sujieren que el tiempo empleado por los varores fue menor en relacion a las mujeres.

4. (1 pt) Descripción de variables numéricas agrupadas por "Age":

In [21]:
data.groupby("Age")['SUS','Time'].describe()

/tmp/ipykernel_8357/2439437661.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  data.groupby("Age")['SUS','Time'].describe()


SUS                                                          Time  \
       count       mean        std   min     25%    50%    75%   max count   
Age                                                                          
forty    6.0  72.083333  14.354152  52.5  63.125  71.25  85.00  87.5   6.0   
thirty   7.0  74.642857  12.535663  57.5  66.250  75.00  83.75  90.0   7.0   
twenty  17.0  78.088235  15.898344  40.0  75.000  80.00  87.50  95.0  17.0   

                                                                     
              mean        std    min     25%    50%      75%    max  
Age                                                                  
forty   177.550000   5.766368  171.0  173.65  176.0  182.325  185.0  
thirty  166.214286  22.118350  128.3  154.65  176.3  181.800  186.0  
twenty  146.835294  20.266350  112.7  135.70  148.0  157.700  188.3

Podemos apreciar que existen tres categorias de edad siendo la mayoria personas  con 20 años.  Respecto a la usabilidad del sistema podemoes aprecia que no existe una diferencia significativa en las mediadas de centralidad, aunque si podemos observar una diferencia en la dedia entre las personas con 40 años y las de 20. Las medidas de dispercion muestran valores similares el unico valor a destacar es el minimo.
Respecto al tiempo vemos que si existe una gran diferencia en la media y la mediana con las personas de 20 años y el grupo de 30 y 40 lo que sugiere que las personas mayores o iguales a 30 años necesitan mayor tiempo para realizar la actividad. Las medidas de despercion muestran iguales resultados a la poblacion mayor o igual a 30 les toma mas tiempo ejecutar el trabajo.


A partir del análsis descriptivo agrupado que ha realizado, construya los siguientes
gráficos: